src/witan/models/dem/ccm/components_functions.clj

```diff
@@ -0,0 +1,17 @@
+(ns witan.models.dem.ccm.components-functions
+  (:require [incanter.core :as i]
+            [clojure.core.matrix.dataset :as ds]
+            [witan.datasets :as wds]))
+
+;; Calculate rates/values for alternative ways to project components of change:
+;; 1) Calculate averages for fixed values/rates or trends:
+(defn calculate-averages
+  "Takes in a dataset with historical data, a number of years to average on
+  and the jumpoff year. Returns a datasets with a new column containing the
+  averages for the estimate column for all the years of interest."
+  [historical-data nb-of-years jumpoff-year]
```

Comment:
Piers said he didn't really use 'jump-off year' but preferred 'base year' with the definition 'the latest year of estimates'. This may be true of other demographer users too, and plan to ask some. I welcome your thoughts, but my inclination is to stick with jump off year for now, defined as 'first year of projection', as it matches the way I'm splitting up functions and the figures I've made.